src/python/WMCore/WorkQueue/WorkQueue.py

```diff
@@ -816,8 +817,29 @@ def closeWork(self, *workflows):
         self.backend.recordTaskActivity('workclosing', msg)
 
         return workflowsToClose
-
-    def performQueueCleanupActions(self, skipWMBS = False):
+    
+    def deleteCompletedWFElements(self):
+        """
+        deletes Workflow when workflow is in finished status
+        """
+        reqNames = self.backend.getWorkflowNames()
+        reqNamesInbox = self.backend.flowNames(inboxFlag = True)
+        reqNames = list(set(reqNames).union(set(reqNamesInbox)))
+        requestsInfo = self.reqmgrSvc.getRequestByNames(reqNames)
```

Comment:
TODO: This need to be more efficient by only returning RequestStatus instead of whole properties.